src/main/java/pl/betoncraft/betonquest/Instruction.java

```diff
@@ -466,12 +466,12 @@ public double getDouble(final String string, final double def) throws Instructio
     }
 
     public ArrayList<Integer> getAllNumbers() {
-        final Pattern p = Pattern.compile("-?\\d+");
+        final Pattern p = Pattern.compile("\\b[\\d]+\\b");
```

Comment:
Useless brackets, how about this regex? `\b(\+|-)?\d+\b` escaped for Java: `\\b(\\+|-)?\\d+\\b`

Reply:
It would be nice to also match + / - but the proposed regex does not do so too except between numbers (e.g. "1-2").

Reply:
- / + seem to count as a \b. That why they are ignored when put before a \d.

Reply:
Ok, after talking with @SaltyAimbOtter i created this regex that should better fit the needs: `(?:\s|\G|^)((\+|-)?\d+)(?:\s|$)`

Escaped for Java: `(?:\\s|\\G|^)((\\+|-)?\\d+)(?:\\s|$)`